arch/risc-v/src/common/supervisor/riscv_context.S

```diff
@@ -0,0 +1,234 @@
+/****************************************************************************
+ * arch/risc-v/src/common/supervisor/riscv_context.S
```

Comment:
can we make supervisor and machine folder contain the same named source files but different implementation? so Make.defs in chip folder could avoid check CONFIG_ARCH_USE_S_MODE as much as possible.

Reply:
> I think having the separate folders is more clear.

Yes, the separated folder looks good.

> There are two files that have the same name and different implementations:
> 
> * riscv_exception_common.S
> * riscv_vectors.S
> 
> Those are selected in Make.defs without reference to the folder. However, there should not be any more common file names.
> 
> I think most of the other files are the .asm functions that are only needed in S-mode

Ok, but let's keep the difference file name as small as possible, so the chip's Make.defs could avoid check CONFIG_ARCH_USE_S_MODE as much as possible.

Reply:
Agreed